### PR TITLE
feat(nexus): nexus snapshot

### PR DESCRIPTION
**Pull request by mtzaurus**
Draft pr to check BDD status / regressions.

### CODE REVIEW

**Comment by mtzaurus** [comment]
bors try

**Comment by mtzaurus** [comment]
bors try

**Comment by mtzaurus** [comment]
bors try

File: io-engine/src/grpc/v0/mayastor_grpc.rs [hunk 1]
```diff
@@ -1741,18 +1745,51 @@ impl mayastor_server::Mayastor for MayastorSvc {
                 let uuid = args.uuid.clone();
                 debug!("Creating snapshot on nexus {} ...", uuid);
```
Comment (hrudaya21) [inline io-engine/src/grpc/v0/mayastor_grpc.rs:1749]: Possibly We need to change the gRPC request name. As We are going to support create_snapshot for both replica and nexus. We may need to revisit the name ?
`create_replica_snapshot`
`create_nexus_snapshot`
Both of these operations can be put part of snapshot.proto and expose to Control plane.
https://github.com/openebs/mayastor-api/pull/63

Comment (mtzaurus) [reply io-engine/src/grpc/v0/mayastor_grpc.rs:1749]: We aren't going to support replica snapshots in v0, only nexus one. In v1 we already have a good separation of API.

Comment (hrudaya21) [reply io-engine/src/grpc/v0/mayastor_grpc.rs:1749]: > We aren't going to support replica snapshots in v0, only nexus one. In v1 we already have a good separation of API.

Ok got it. What about changing name in `v1/nexus.rs` ?

**Review by hrudaya21 — APPROVED** [review]

**Review by hrudaya21 — APPROVED** [review]

**Review by dsavitskiy — APPROVED** [review]

**Comment by mtzaurus** [comment]
bors merge

**Pull request merged**